KIPs/kip-xxx.md

```diff
@@ -0,0 +1,178 @@
+```
+
+Now both EOAs and SCAs hold the same set of 7 fields: `nonce`, `balance`, `humanReadable`, `accountKey`, `storageRoot`, `codeHash`, and `codeInfo`.
+
+When an EOA is RLP encoded, the compact encoding is used when the account has storageRoot = EMPTY_ROOT_HASH, codeHash = EMPTY_CODE_HASH, and codeInfo = 0x0. Conversely, when decoding an RLP-encoded EOA, the compact encoding implies that the account has storageRoot = EMPTY_ROOT_HASH, codeHash = EMPTY_CODE_HASH, and codeInfo = 0x0.
+
+If an EOA’s delegation was cleared by a SetCode transaction with authorization.address = 0x0, set its CodeHash to EMPTY_CODE_HASH as per EIP-7702, and also set its CodeInfo to 0x0. Note that the account does not necessarily return to compact encoding. It will be compactly encoded only if storageRoot is EMPTY_ROOT_HASH.
+
+An EOA with codeHash = EMPTY_CODE_HASH can be called “EOA without code” and an EOA with codeHash != EMPTY_CODE_HASH can be called “EOA with code”. Note that EOA without code doesn’t always mean it will be compactly encoded.
+
+### Interaction with other transaction types
+
+The `to` address of the following transaction types MUST be an EOA without code:
+
+- `TxTypeValueTransfer`
+- `TxTypeFeeDelegatedValueTransfer`
+- `TxTypeFeeDelegatedValueTransferWithRatio`
+- `TxTypeValueTransferMemo`
+- `TxTypeFeeDelegatedValueTransferMemo`
+- `TxTypeFeeDelegatedValueTransferMemoWithRatio`
+
+The `from` address of the following transaction types MUST be an EOA without code:
+
+- `TxTypeAccountUpdate`
+- `TxTypeFeeDelegatedAccountUpdate`
+- `TxTypeFeeDelegatedAccountUpdateWithRatio`
+
+The `to` address of the following transaction types MUST be an EOA with code or an SCA:
+
+- `TxTypeSmartContractExecution`
+- `TxTypeFeeDelegatedSmartContractExecution`
+- `TxTypeFeeDelegatedSmartContractExecutionWithRatio`
+
+Note that whether the emptiness of storageRoot does not matter in transaction type restrictions.
+
+### Changes to APIs
+
+#### eth_getRawTransaction, eth_sendRawTransaction
+
+eth_getRawTransaction returns the raw transaction RLP prefixed with `SET_CODE_TX_TYPE (0x04)` and eth_sendRawTransaction accepts such raw transaction.
+
+#### kaia_getRawTransaction, kaia_sendRawTransaction
+
+kaia_getRawTransaction returns the raw transaction RLP prefixed with `ENVELOPED_SET_CODE_TX_TYPE (0x7804)` and kaia_sendRawTransaction accepts such raw transaction.
+
+#### eth_getBlock*, eth_getTransaction*, eth_getTransactionReceipt
+
+For SetCode transaction type, eth namespace APIs that return the transaction fields MUST show the `authorizationList` field and its `type: "0x04"`.
+
+#### kaia_getBlock*, kaia_getTransaction*, kaia_getTransactionReceipt
+
+For SetCode transaction type, kaia namespace APIs that return the transaction fields MUST show the `authorizationList` field and its `typeInt: 30724`, and `type: "TxTypeEthereumSetCode"`.
+
+#### eth_getCode, eth_getStorageAt, kaia_getCode, kaia_getStorageAt
```

Comment:
As I commented in [#12](https://github.com/kaiachain/kaia/pull/126#discussion_r1847979063), for compatibility reasons, I think it would be fine to keep it as it is, returning false if the codeHash is empty even if it is a ContractAccount.

Reply:
Agree. Also the codehash criteria  coincides with the API description. Let me add in the Backwards Compatibility for clarification.

Reply:
Added 58e7bcd.

Reply:
Returning true for EOA and SCA seems fine, but should we keep the names(kaia_isContractAccount)?

Reply:
Yes, we keep the name for backwards compatibility. I personally want to remove this API because eth_getCode and kaia_getAccount can do the same thing. But since we don't know how this API is used, we cannot just remove or rename this API.